### PR TITLE
Simple strategist

### PR DESCRIPTION
**Pull request by robertecurtin**
This module will be responsible for aggregating the meta-sensor information and selecting which tactic should be employed.

Tactics modules will be used to implement the behavior within a given tactic.

Meta-sensor modules will take the sensor inputs and interpret them into higher-level information about the surrounding world, which can then be provided to the strategist and/or tactics modules.

### CODE REVIEW

File: src/app/modules/meta-sensing/enemy_location.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+typedef uint8_t enemy_location_t;
+
+enum {
+  enemy_location_unknown,
+  enemy_location_front_center,
+  enemy_location_front_left,
+  enemy_location_front_right
+};
+
+#endif
```
Comment (ryanplusplus) [inline src/app/modules/meta-sensing/enemy_location.h]: You're killing me, smalls

File: src/app/modules/meta-sensing/enemy_location.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#include <stdint.h>
+
+typedef uint8_t enemy_location_t;
+
+enum {
+  enemy_location_unknown,
+  enemy_location_front_center,
+  enemy_location_front_left,
+  enemy_location_front_right
+};
```
Comment (ryanplusplus) [inline src/app/modules/meta-sensing/enemy_location.h]:
```suggestion
};
typedef uint8_t enemy_location_t;
```

File: src/app/modules/tactics/strategist.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  tiny_fsm_t fsm;
+  i_tiny_key_value_store_t* key_value_store;
+  const strategist_keys_t* keys;
+} strategist_t;
+
+void strategist_init(
+  strategist_t* self,
+  i_tiny_key_value_store_t* key_value_store,
+  const strategist_keys_t* keys);
+#endif
```
Comment (ryanplusplus) [inline src/app/modules/tactics/strategist.h]:
```suggestion

#endif
```

File: test/strategist_test.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+enum {
+  in_front = enemy_location_front_center,
+  not_visible = enemy_location_unknown
+};
+
+// clang-format off
+  #define data_model_key_value_pairs(pair) \
+    pair(key_tactic,            tactic_t) \
+    pair(key_enemy_location,    enemy_location_t)
```
Comment (ryanplusplus) [inline test/strategist_test.cpp]: This appears to have an extra indent

File: src/app/modules/meta-sensing/tactic.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#define tactic_h
+
+#include <stdint.h>
+
+typedef uint8_t tactic_t;
+
+enum {
+  tactic_seeking,
+  tactic_charge
+};
```
Comment (ryanplusplus) [inline src/app/modules/meta-sensing/tactic.h]:
```suggestion
};
typedef uint8_t tactic_t;
```

File: src/app/modules/meta-sensing/enemy_location.h (hunk 1 above)
Comment (robertecurtin) [reply src/app/modules/meta-sensing/enemy_location.h]: blame the formatter

File: src/app/modules/meta-sensing/enemy_location.h (hunk 2 above)
Comment (robertecurtin) [reply src/app/modules/meta-sensing/enemy_location.h]: was following your lead from the other files :trifarce:

Comment (ryanplusplus) [reply src/app/modules/meta-sensing/enemy_location.h]: Must have been a derp.

File: src/app/modules/tactics/strategist.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  tiny_key_value_store_key_t key_robot_running;
+  tiny_key_value_store_key_t key_line_detected;
+  tiny_key_value_store_key_t key_tactic_stopped;
+} strategist_keys_t;
+
+typedef struct {
+  tiny_event_subscription_t on_change_subscription;
+  tiny_fsm_t fsm;
+  i_tiny_key_value_store_t* key_value_store;
+  const strategist_keys_t* keys;
```
Comment (ryanplusplus) [inline src/app/modules/tactics/strategist.h]: Is there any reason these guys are not `_private`?

Comment (robertecurtin) [reply src/app/modules/tactics/strategist.h]: at ease

**Pull request merged**